Board.java

```diff
@@ -16,11 +16,10 @@ public String getState() {
     }
 
      public Boolean availablePosition(int position) {
-        if ( position >= 0 && position < dimension && state.charAt(position) == '-') {
-            return true;
-        }
-        System.out.println("Not a valid position");
-        return false;
+         if ( position >= 0 && position <= dimension && state.charAt(position) == '-') {
+             return true;
+         }
+         return false;
```

Comment:
this whole block can be reduced to:
position >= 0 && position <= dimension && state.charAt(position) == '-'